modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -18,18 +18,76 @@ import * as Parents from '../dom/Parents';
 import * as TableCellSelection from '../selection/TableCellSelection';
 import * as DeleteElement from './DeleteElement';
 import * as TableDeleteAction from './TableDeleteAction';
+import * as TableDeleteUtils from './TableDeleteUtils';
 
 const freefallRtl = (root: SugarElement<Node>): Optional<SugarElement<Node>> => {
   const child = SugarNode.isComment(root) ? Traverse.prevSibling(root) : Traverse.lastChild(root);
   return child.bind(freefallRtl).orThunk(() => Optional.some(root));
 };
 
-const emptyCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[]) => {
+const shallowCopyRange = (rng: Range) => ({
+  startContainer: rng.startContainer,
+  startOffset: rng.startOffset,
+  endContainer: rng.endContainer,
+  endOffset: rng.endOffset
+});
+
+const removeContentFromCells = (cells: SugarElement<HTMLTableCellElement>[]) =>
   Arr.each(cells, PaddingBr.fillWithPaddingBr);
-  editor.selection.setCursorLocation(cells[0].dom, 0);
+
+/*
+ * Runs when
+ * - the start and end of the selection is contained within the same table (called directly from deleteRange)
+ * - part of a table and content outside is selected
+ */
+const emptySingleTableCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], outsideRng: Optional<Range>) => {
+  const selection = editor.selection;
+  const rng = selection.getRng();
+  const { startContainer, startOffset, endContainer } = shallowCopyRange(rng);
+  const { startInTable, partialSelection } = TableDeleteUtils.getTableDetailsFromRange(rng, SugarElement.fromDom(editor.getBody()));
+  const outsideBlock = Optionals.someIf(partialSelection, SugarElement.fromDom(editor.dom.getParent(startInTable ? endContainer : startContainer, editor.dom.isBlock)));
```

Comment:
This does a reasonable amount of computation when not needed, so this would be better as a ternary. We can also move the filter call here since we never use it for anything else.